jina/executors/__init__.py

```diff
@@ -351,18 +352,27 @@ def config_abspath(self) -> str:
 
     @staticmethod
     def get_shard_workspace(
-        workspace_folder: str, workspace_name: str, pea_id: int
+        workspace_folder: str,
+        workspace_name: str,
+        pea_id: int,
+        replica_id: Optional[int] = None,
     ) -> str:
         """
         Get the path of the current shard.
 
-        :param workspace_folder: Folder of the workspace.
-        :param workspace_name: Name of the workspace.
-        :param pea_id: Id of the pea,
+        :param workspace_folder: folder of the workspace.
+        :param workspace_name: name of the workspace.
+        :param pea_id: id of the pea
+        :param replica_id: id of the replica
 
         :return: returns the workspace of the shard of this Executor.
         """
-        return os.path.join(workspace_folder, f'{workspace_name}-{pea_id}')
+        if replica_id is None or replica_id == 'None':
```

Comment:
This should be checked in a test, right?

Reply:
Since `replica_id` is `Optional[int]`, how can it ever be a string?

Reply:
@cristianmtr I created a test for the workspace
@maximilianwerk after setting it as None the type got lost and then in some scenarios, it was interpreted as String. Instead of None, I use now -1. We do the same for the pea_id. It feels not so clean but works.